### ISSUE TITLE
Feature Request: running natively on Windows 10

### ISSUE THREAD

**Issue by aont** [tmux/tmux#1954]
On Windows, tmux from Cygwin or MSYS2 is available.
But, it is unable to use tmux on native console (on cmd.exe).
By the way, recent versions of Windows 10 have ConPTY API.
Thanks to ConPTY API, running tmux on WSL is available.
So I think, running tmux natively on Windows 10 with ConPTY API is available.
Is this right ?

**Comment by nicm** [comment]
I don't intend to port tmux to Windows but as always if you send patches I will review them.

**Comment by aont** [comment]
I see. Since it is difficult for me to code patches, I will close this request.

**Comment by krader1961** [comment]
@aont, tmux should "just work" on WSL. If it does not work on WSL it is likely to be due to a bug in the WSL emulation of UNIX behavior. I would also expect it to work in a Cygwin based terminal but that is less likely due to the quirks of Cygwin which means it does not fully and faithfully provide a UNIX like environment. A failure could also be due to a bug in tmux but that is far less likely. Using tmux in a `cmd.exe` session is unlikely to ever be supported because the semantics for Windows native apps are very different from classic UNIX or POSIX 1003.

In other words, if you find that tmux fails to work as expected in WSL you should probably open an issue here with appropriate details of the failure. If, on the other hand, you are asking that tmux work in a Windows native cmd.exe session that will probably never happen.

**Comment by aont** [comment]
Thank you for the kind reply. I am asking the latter one.
> the semantics for Windows native apps are very different from classic UNIX or POSIX 1003.

Yes, I understand the difficulty. I just thought that since WSL runs on Windows's native console, it is possible to run tmux without WSL, though that needs patches for tmux (or cygwin layer).
